src/listmode_buildblock/CListModeDataECAT8_32bit.cxx

```diff
@@ -46,9 +46,19 @@ CListModeDataECAT8_32bit(const std::string& listmode_filename)
 
   this->exam_info_sptr.reset(new ExamInfo(*interfile_parser.get_exam_info_ptr()));
 
-
-  this->proj_data_info_sptr = interfile_parser.data_info_ptr;
-  this->scanner_sptr.reset( new Scanner(*this->proj_data_info_sptr->get_scanner_ptr()));
+  const std::string originating_system(this->interfile_parser.get_exam_info_ptr()->originating_system);
+  shared_ptr<Scanner> this_scanner_sptr(Scanner::get_scanner_from_name(originating_system));
+  if (this_scanner_sptr->get_type() == Scanner::Unknown_scanner)
+    error(boost::format("Unknown value for originating_system keyword: '%s") % originating_system );
+
+  std::unique_ptr<ProjDataInfo> tmp_unique(ProjDataInfo::construct_proj_data_info(this_scanner_sptr,
```

Comment:
these lines look quite problematic. I know you put it in to circumvent compilation problems, but I cannot understand why it works.

We cannot use `std::unique` explicitly as it will fail on pre-C++11 compilers. (we have some ugly work-arounds for that).

from current InterfileHeaderSiemens, it looks like we should be able to assign the result of `construct_proj_data_info` diractly to a `shared_ptr` anyway.

In any case, i don't understand why this needs to be changed from the original. all the work is done in `InterfileRawDataHeaderSiemens`